src/software/ai/hl/stp/play/indirect_free_kick_play.cpp

```diff
@@ -0,0 +1,293 @@
+{
+    double min_dist_to_corner =
+        std::min((world.field().enemyCornerPos() - world.ball().position()).len(),
+                 (world.field().enemyCornerNeg() - world.ball().position()).len());
+
+    // Make sure we don't interfere with the cornerkick play
+    return world.gameState().isOurFreeKick() &&
+           min_dist_to_corner >= CornerKickPlay::BALL_IN_CORNER_RADIUS;
+}
+
+bool IndirectFreeKickPlay::invariantHolds(const World &world) const
+{
+    return (world.gameState().isPlaying() || world.gameState().isReadyState()) &&
+           (!Evaluation::teamHasPossession(world, world.enemyTeam()) ||
+            Evaluation::teamPassInProgress(world, world.friendlyTeam()));
+}
+
+void IndirectFreeKickPlay::getNextTactics(TacticCoroutine::push_type &yield)
+{
+    /**
+     * This play is basically:
+     * - One robot attempts to pass, and chips towards the enemy goal if it can't
+     *   find a pass in time
+     * - Two robots try to get in good positions in the enemy end to receive a pass
+     * - Two robots crease defend
+     * - One robot is goalie
+     */
+
+    // Setup the goalie
+    auto goalie_tactic = std::make_shared<GoalieTactic>(
+        world.ball(), world.field(), world.friendlyTeam(), world.enemyTeam());
+
+    // Setup crease defenders to help the goalie
+    std::array<std::shared_ptr<CreaseDefenderTactic>, 2> crease_defender_tactics = {
+        std::make_shared<CreaseDefenderTactic>(world.field(), world.ball(),
+                                               world.friendlyTeam(), world.enemyTeam(),
+                                               CreaseDefenderTactic::LeftOrRight::LEFT),
+        std::make_shared<CreaseDefenderTactic>(world.field(), world.ball(),
+                                               world.friendlyTeam(), world.enemyTeam(),
+                                               CreaseDefenderTactic::LeftOrRight::RIGHT),
+    };
+
+    // If the passing is coming from the friendly end, we split the cherry-pickers
+    // across the x-axis in the enemy half
+    Rectangle cherry_pick_1_target_region(world.field().centerPoint(),
+                                          world.field().enemyCornerPos());
+    Rectangle cherry_pick_2_target_region(world.field().centerPoint(),
+                                          world.field().enemyCornerNeg());
+
+    // Otherwise, the pass is coming from the enemy end, put the two cherry-pickers
+    // on the opposite side of the x-axis to wherever the pass is coming from
+    if (world.ball().position().x() > -1)
+    {
+        double y_offset =
+            -std::copysign(world.field().xLength() / 2, world.ball().position().y());
+        cherry_pick_1_target_region =
+            Rectangle(Point(0, world.field().xLength() / 4),
+                      Point(world.field().xLength() / 2, y_offset));
+        cherry_pick_2_target_region =
+            Rectangle(Point(0, world.field().xLength() / 4), Point(0, y_offset));
```

Comment:
I think divide by double would look better
```suggestion
            Rectangle(Point(0, world.field().xLength() / 4.0), Point(0, y_offset));
```